posthog/api/web_vitals.py

```diff
@@ -0,0 +1,79 @@
+from rest_framework import viewsets, exceptions, status
+from rest_framework.response import Response
+from rest_framework.request import Request
+from posthog.api.routing import TeamAndOrgViewSetMixin
+from drf_spectacular.utils import extend_schema, OpenApiParameter
+from drf_spectacular.types import OpenApiTypes
+
+from posthog.hogql_queries.query_runner import get_query_runner, ExecutionMode
+
+
+class WebVitalsViewSet(TeamAndOrgViewSetMixin, viewsets.ReadOnlyModelViewSet):
+    scope_object = "web_vitals"
```

Comment:
Let's instead use the existing `scope_object = "query"` value, since this endpoint just boils down to a query which can be used with `TemporaryTokenAuthentication`.

But wait – it doesn't looks like `TemporaryTokenAuthentication` is actually allowed on this viewset, so this would in fact not work for PostHog users (it only works with PostHog in local dev's self-capture mode)

Reply:
Ok, good catch, I wasn't aware of that

I've moved to using `query` here and added a new authentication class to allow the temporary token to work